algorithms/2-search-solutions/binary_search_iterative.rb

```diff
@@ -0,0 +1,17 @@
+def binary_search(sorted_array, value)
+  collection_size = sorted_array.length - 1
+  low = 0
+  high = collection_size
+
+  while low <= high
+    mid = (low + high) / 2
+    if sorted_array[mid] > value
+      high = mid - 1
+    elsif sorted_array[mid] < value
+      low = mid + 1
+    else
+      return mid
+    end
+  end
+  return "Value not found."
```

Comment:
It is usually better to use a consistent type when returning. Here, either `nil` or `-1` are ones I have seen. You could also throw an exception, which is my favorite way to handle it.